src/pages/export/Data.js

```diff
@@ -36,46 +43,101 @@ export class DataExport extends FormBase {
     formWidth = 800
     formTitle = i18n.t('Data Export')
     submitLabel = i18n.t('Export')
+    fields = []
+    state = {
+        _meta: {
+            submitted: false,
+            valid: false,
+            processing: false,
+            error: null,
+        },
+    }
 
-    fields = [
-        ...getFormFields([
-            'orgUnit',
-            'children',
-            'selectedDataSets',
-            'startDate',
-            'endDate',
-            'format',
-            'compression',
-        ]),
-
-        getFormFieldMoreOptions(),
-
-        ...getFormFields([
-            'includeDeleted',
-            'dataElementIdScheme',
-            'orgUnitIdScheme',
-            'categoryOptionComboIdScheme',
-        ]),
-    ]
-
-    state = getFormValues([
-        'orgUnit',
-        'children',
-        'selectedDataSets',
-        'startDate',
-        'endDate',
-        'format:.json:json,xml,csv',
-        'compression',
-        'includeDeleted',
-        'dataElementIdScheme',
-        'orgUnitIdScheme',
-        'categoryOptionComboIdScheme',
-    ])
+    setFormValues() {
+        this.setState({
+            ...getFormValues([
+                'orgUnit',
+                'children',
+                'selectedDataSets',
+                'startDate',
+                'endDate',
+                'format:.json:json,xml,csv',
+                'compression',
+                'includeDeleted',
+                'dataElementIdScheme',
+                'orgUnitIdScheme',
+                'categoryOptionComboIdScheme',
+            ]),
+        })
+    }
 
     async componentDidMount() {
+        this.props.fetchDataElementAttributes()
+        this.props.fetchOrganisationUnitAttributes()
         await this.fetch()
     }
 
+    initializeFormValues(fieldValuesOverride) {
```

Comment:
I think this method, and componentDidUpdate could use some comments on what's going on exactly. They're the areas where it's not clear to me what's happening and why.

Reply:
I've realized I don't have to set the fields and state after the options have been loaded, so I just reverted that. I've added comments to componentDidUpdate